lib/commands/skill/add-locales/helper.js

```diff
@@ -0,0 +1,148 @@
+        const iModelFolderPath = path.join(process.cwd(), ResourcesConfig.getInstance().getSkillMetaSrc(profile),
+            CONSTANTS.FILE_PATH.SKILL_PACKAGE.INTERACTION_MODEL, 'custom');
+        // copy iModel JSONs that contains same language
+        iModelSourceByLocales.forEach((v, k) => {
+            if (fs.existsSync(v)) {
+                const sourceLocale = path.basename(v, path.extname(v));
+                const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                fs.copySync(v, targetFilePath); // do not fail if the target locale exists already
+                uniqueTemplateSet.delete(v);
+                const sourceManifestLocale = Manifest.getInstance().getPublishingLocale(sourceLocale);
+                Manifest.getInstance().setPublishingLocale(k, sourceManifestLocale);
+            }
+        });
+        // retrieve remote templates when no existing JSON shares the same language
+        _retrieveTemplatesByLanguage(uniqueTemplateSet, doDebug, (templateErr, templateByLanguage) => {
+            if (templateErr) {
+                return callback(templateErr);
+            }
+            iModelSourceByLocales.forEach((v, k) => {
+                if (!fs.existsSync(v)) {
+                    const targetFilePath = path.join(iModelFolderPath, `${k}.json`);
+                    fs.writeFileSync(targetFilePath, templateByLanguage.get(v), { encoding: 'utf-8' });
+                    Manifest.getInstance().setPublishingLocale(k, { name: 'please change' });
+                }
+            });
+            Manifest.getInstance().write();
+            callback(undefined, iModelSourceByLocales);
+        });
+    });
+}
+
+/**
+ * Based on user's selected locales, and the local project's iModel files, decide where to get the
+ * iModel file to add in user's local project. Exclude reusableLocales themselves from the result.
+ *
+ * @param {Array} selectedLocales
+ * @param {String} profile
+ * @param {Boolean} doDebug
+ * @returns {Function} callback Map { locale: filePath/remoteURI }
+ */
+function _getNewLocaleModelUri(selectedLocales, profile, doDebug, callback) {
+    const skillMetaController = new SkillMetadataController({ profile, doDebug });
+    const localIModelByLocale = skillMetaController.getInteractionModelLocales();
+    const result = new Map();
+    // get from map of templates
+    httpClient.request({
+        url: CONSTANTS.TEMPLATES.INTERACTION_MODEL_MAP,
+        method: CONSTANTS.HTTP_REQUEST.VERB.GET
+    }, 'GET_INTERACTION_MODEL_MAP', doDebug, (error, templateMapResponse) => {
+        if (error || !templateMapResponse.statusCode || templateMapResponse.statusCode !== 200) {
+            return callback('Failed to retrieve the template list. Please run again with --debug to check more details.');
+        }
+        let templateIndexMap = templateMapResponse.body;
+        if (typeof templateMapResponse.body === 'string') {
```

Comment:
why do you need this check?

Reply:
in case the s3 gets uploaded in the wrong format: uploaded as a JSON string, the code can still handle

Reply:
currently it is a JSON string. Was it uploaded incorrectly?

Reply:
The response is JSON or not is dependent on the configuration in the "request" module. If user set `{ json: true }` in the request option, the result.body will be JSON.

Because I copied this pattern from another file, I have corrected both places to always take the response.body as a JSON string, and parse when received.

Reply:
If I understand it right, there is option in http client to parse response body as json. I think you should pass the option instead of having this if check.